### PR TITLE
New service onboarding for Microsoft.EnterpriseSupport

### PR DESCRIPTION
**Pull request by dasaevv555**
# ARM (Control Plane) API Specification Update Pull Request 

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I have reviewed the general guidance on the spec PR review process: https://aka.ms/specprreview.
- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data-plane related specifications.
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy).
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can request review from an ARM API Review board.

### ARM API changes review

- If you want for the ARM team to review this PR, you must add the `ARMReview` label. 
- The automation may automatically add the `ARMReview` label, if appropriate.  
  If this happens, proceed according to guidance given in GitHub comments also added by the automation.

### Breaking change review

If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), 
follow the process outlined in the [High-level Breaking Change Process doc](https://aka.ms/breakingchangesprocess#high-level-breaking-change-process).
      
## Getting help

- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For additional help, see https://aka.ms/azsdk/support/spectools.

### CODE REVIEW

File: specification/EnterpriseSupport/resource-manager/Microsoft.EnterpriseSupport/preview/2023-05-01-preview/EnterpriseSupport.json [hunk 1]
```diff
@@ -0,0 +1,755 @@
...
+          "type": "string",
+          "description": "BicCat product ID"
+        },
+        "skuId": {
+          "type": "string",
+          "description": "BigCat SKU ID"
+        },
+        "availabilityId": {
+          "type": "string",
+          "description": "BigCat availability ID"
```
Comment (wilcobmsft) [inline specification/EnterpriseSupport/resource-manager/Microsoft.EnterpriseSupport/preview/2023-05-01-preview/EnterpriseSupport.json:521]: We should avoid exposing BigIds externally... Ideally productCode (a UPN) is the only type of identifier we expose right now.

File: specification/EnterpriseSupport/resource-manager/Microsoft.EnterpriseSupport/preview/2023-05-01-preview/EnterpriseSupport.json [hunk 2]
```diff
@@ -0,0 +1,755 @@
...
+        },
+        "billingAccountId": {
+          "type": "string",
+          "description": "Billing Account ID"
+        },
+        "billingGroupId": {
+          "type": "string",
+          "description": "Billing Group ID"
+        },
+        "projectId": {
```
Comment (wilcobmsft) [inline specification/EnterpriseSupport/resource-manager/Microsoft.EnterpriseSupport/preview/2023-05-01-preview/EnterpriseSupport.json:507]: We should not need billingAccountId/bgId/projectId... All of that we should be able to infer from the subscription scope. If you call Create Asset directly, you can just pass in the subscriptionId directly, and Assets will resolve all of this automatically.

File: specification/EnterpriseSupport/resource-manager/Microsoft.EnterpriseSupport/preview/2023-05-01-preview/EnterpriseSupport.json [hunk 3]
```diff
@@ -0,0 +1,755 @@
...
+        },
+        "quantity": {
+          "description": "Quantity for the purchase"
+        },
+        "serviceStartDate": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Service start date for  Enterprise Support"
+        },
+        "csam": {
```
Comment (wilcobmsft) [inline specification/EnterpriseSupport/resource-manager/Microsoft.EnterpriseSupport/preview/2023-05-01-preview/EnterpriseSupport.json:535]: Why do we need this? I don't believe we do this in any other RP today which is used by Quote...

File: specification/EnterpriseSupport/resource-manager/Microsoft.EnterpriseSupport/preview/2023-05-01-preview/EnterpriseSupport.json [hunk 4]
```diff
@@ -0,0 +1,755 @@
...
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "ListOperations": {
+            "$ref": "./examples/Operations_Get.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.EnterpriseSupport/EnterpriseSupports": {
```
[the remark this answers is not in the document]
Comment (kuremichi) [reply specification/EnterpriseSupport/resource-manager/Microsoft.EnterpriseSupport/preview/2023-05-01-preview/EnterpriseSupport.json:88]: this was the preference from our PM. 

Below will be some sample resource types under this namespace

//current
providers/Microsoft.EnterpriseSupport/EnterpriseSupports

// future
providers/Microsoft.EnterpriseSupport/AzureProDirects
providers/Microsoft.EnterpriseSupport/OtherSupportOfferingA
providers/Microsoft.EnterpriseSupport/OtherSupportOfferingB

File: specification/EnterpriseSupport/resource-manager/Microsoft.EnterpriseSupport/preview/2023-05-01-preview/EnterpriseSupport.json (hunk 1 above)
Comment (kuremichi) [reply specification/EnterpriseSupport/resource-manager/Microsoft.EnterpriseSupport/preview/2023-05-01-preview/EnterpriseSupport.json:521]: we are going to release this week with P/S/A/T and UPN in the data contract. we will come back to revise this in Aug/Sep timeframe to drop PSAT.

File: specification/EnterpriseSupport/resource-manager/Microsoft.EnterpriseSupport/preview/2023-05-01-preview/EnterpriseSupport.json (hunk 2 above)
Comment (kuremichi) [reply specification/EnterpriseSupport/resource-manager/Microsoft.EnterpriseSupport/preview/2023-05-01-preview/EnterpriseSupport.json:507]: this lookup of BA/BG can be achieved by calling the bigcat hydrate call using the upn. the info will be available under x-ms-bst

File: specification/EnterpriseSupport/resource-manager/Microsoft.EnterpriseSupport/preview/2023-05-01-preview/EnterpriseSupport.json (hunk 3 above)
Comment (kuremichi) [reply specification/EnterpriseSupport/resource-manager/Microsoft.EnterpriseSupport/preview/2023-05-01-preview/EnterpriseSupport.json:535]: CSAM stands for Customer satisfaction account manager, or relationship manager. This is a microsoft personnel performing the delivery role. In today's business process, this is determined and captured in the sales process today. For launching enterprise support on MCA, this existing practice is going to be carried over.

File: specification/EnterpriseSupport/resource-manager/Microsoft.EnterpriseSupport/preview/2023-05-01-preview/EnterpriseSupport.json (hunk 2 above)
Comment (kuremichi) [reply specification/EnterpriseSupport/resource-manager/Microsoft.EnterpriseSupport/preview/2023-05-01-preview/EnterpriseSupport.json:507]: we will treat this as a change and update the arm contract in August/September timeframe.

**Pull request closed without merging**